webanno/webanno
# #1404 - Modify slot behavior for Annotations other than spans

**What's in the PR**
- Introduce new event interfaces to allow generically listening to all annotation creation or deletion evens
- allow ADEP.actionFillSlot() to be successful even if the armed slot is NOT in the ADEP and even if the link host is NOT an AnnotationFS
- Pass the entire feature state along when arming a slot to allow direct access to it when the slot is actually filled later on
- new RenderSlotsEvent is fired by AnnotatorStateImpl when setArmedSlot() is called


**How to test manually**
Do the following steps in INCEpTION:
1. Create a new document annotation layer with a link feature to a span annotation layer.
2. Open a document
3. Create a document annotation with the type of your new layer
4. Select the new slot.
5. Annotate something in the text.
6. Link should be created

Also try out having a span layer with a link to another span.
Then have a document annotation and a span annotation both with links open and play around to see if the slots are armed/disarmed correctly.

**Automatic testing**

**Documentation**
